backend/src/main/java/ai/verta/modeldb/batchProcess/CollaboratorResourceMigration.java

```diff
@@ -297,6 +301,7 @@ private static void migrateRepositories() {
               }
             }
           }
+          deleteRoleBindingsOfRepositories(migratedRepositoryEntities);
```

Comment:
This needs to be inside the migration step. We cannot set `repository.setVisibility_migration(true);` and then have this. It's possible to have a failure between these steps and the role binding will never be removed, leading to inconsistencies. We need to make our migrations safe.

Reply:
updated

Reply:
This is not done before `repository.setVisibility_migration(true);`. The code has the same issue as before: I can mark the migration as completed before it's completed. We can only set the migration as done if every single operation we'll do for that given element is done and successful. If they are not, we *cannot* set the migration as done.

Reply:
Sorry, my bad. Updated again

Reply:
Thanks! It's a small change, but it's good for us to develop this muscle memory.